### PR TITLE
handle username and passwords correctly with base64 encoding

### PR DESCRIPTION
**Pull request by garrensmith**

### CODE REVIEW

File: app/addons/replication/api.js [hunk 1]
```diff
@@ -41,6 +42,38 @@ export const getAuthHeaders = (username, password) => {
...
+      username: '',
+      password: ''
+    };
+  }
+
+  const startIndex = url.startsWith("https") ? 8 : 7;
+  const rawCreds = url.slice(startIndex, index);
+  const colonIndex = rawCreds.indexOf(':');
+  const username = rawCreds.slice(0, colonIndex);
+  const password = rawCreds.slice(colonIndex + 1, rawCreds.length);
```
Comment (millayr) [inline app/addons/replication/api.js]: Minor note but I think you could combine all of these variable instantiations and assignments in to a single const statement.

File: app/addons/replication/api.js [hunk 2]
```diff
@@ -41,6 +42,38 @@ export const getAuthHeaders = (username, password) => {
...
+  };
+};
+
+export const removeCredentialsFromUrl = (url) => {
+  const index = url.lastIndexOf('@');
+  if (index === -1) {
+    return url;
+  }
+
+  const protocal = url.startsWith("https") ? "https://" : 'http://';
```
Comment (millayr) [inline app/addons/replication/api.js]: protoc_o_l :)

**Comment by millayr** [comment]
+1

**Comment by garrensmith** [comment]
merged

**Pull request closed without merging**